lib/src/main/java/com/auth0/jwt/JWTVerifier.java

```diff
@@ -353,7 +353,7 @@ private void verifyClaimValues(DecodedJWT jwt, Map.Entry<String, Object> expecte
 
     private void assertClaimPresent(Claim claim, String claimName) {
         if (claim instanceof NullClaim) {
-            throw new InvalidClaimException(String.format("The Claim '%s' is not present in the JWT.", claimName));
+            throw new InvalidClaimException(String.format("The Claim '%s' is not present in the JWT.", claimName), claimName);
```

Comment:
I like the goal that you are trying to achieve with this PR but I'm not convinced of usages like this one, in which the `claimName` value has to be passed twice in the same line. I assume the focus was on trying not to make any breaking changes.

Ideally, we would have two different exception classes. One that could represent a "missing claim" and one that could represent an "unexpected claim". Then those constructors would take just the claim name and the message would be set internally in there. I think this is something doable today also without breaking changes by subclassing the `InvalidClaimException`. Let me check with @jimmyjames what he thinks about adding two more exceptions to improve this use case.

Reply:
You're right. I tried to not break the current API. I can provide an other PR when you'll agree with a new API.